packages/ddp-server/writefence.js

```diff
@@ -59,6 +60,15 @@ _.extend(DDPServer._WriteFence.prototype, {
     self._maybeFire();
   },
 
+  // Register a function to be called before firing the fence.
+  onBeforeFire: function (func) {
+    var self = this;
+    if (self.fired)
```

Comment:
In fact, this makes some of the oplog logic a little easier to understand.  Because you clean up the `_oplogObserveDrivers` field at the end of the `onBeforeFire` callback, I could worry that somehow another observe driver could come along and register itself again.  However, if we ensure that you can't add a second `onBeforeFire` callback after the fence is armed, and the fence must be armed for the first callback to run, we know we'll at least see an ugly error if this happens.